### PR TITLE
report(fr): add tsx to shell report

### PR DESCRIPTION
**Pull request by adamraine**
`standalone-flow.js` -> `standalone-flow.tsx`

Will merge into https://github.com/GoogleChrome/lighthouse/pull/12850 before landing.

Ref #11313

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -168,6 +168,7 @@
     "webtreemap-cdt": "^3.2.1"
   },
   "dependencies": {
+    "@rollup/plugin-typescript": "^8.2.5",
```
Comment (connorjclark) [inline package.json:171]: devdep

File: report/clients/standalone-flow.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * @fileoverview The entry point for rendering the Lighthouse report for the HTML file created by ReportGenerator.
+ * The renderer code is bundled and injected into the report HTML along with the JSON report.
+ */
+
+import {render, FunctionComponent} from 'preact';
+import {useState} from 'preact/hooks';
+
+/* global window document */
+
+// eslint-disable-next-line no-undef
```
Comment (adamraine) [inline report/clients/standalone-flow.tsx:18]: TODO: Update linter rules

File: report/clients/standalone-flow.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/**
```
Comment (brendankenny) [inline report/clients/standalone-flow.tsx:1]: probably makes sense in `report/clients/flow/*`, especially if it's going to have its own tsconfig, eslint rules, etc

Comment (connorjclark) [reply report/clients/standalone-flow.tsx:1]: maybe we should do a new folder `/flow`? toplevel.

Comment (brendankenny) [reply report/clients/standalone-flow.tsx:1]: > maybe we should do a new folder `/flow`? toplevel

yeah, it's different enough that makes sense too

Comment (adamraine) [reply report/clients/standalone-flow.tsx:1]: I like toplevel

Comment (connorjclark) [reply report/clients/standalone-flow.tsx:1]: in that case I'd prefer `flow-report/` fyi

**Review by patrickhulce — COMMENTED** [review]
thanks for spinning this up! :)

File: flow-report/standalone-flow.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ * @fileoverview The entry point for rendering the Lighthouse report for the HTML file created by ReportGenerator.
+ * The renderer code is bundled and injected into the report HTML along with the JSON report.
+ */
+
+import {render, FunctionComponent} from 'preact';
+import {useState} from 'preact/hooks';
+
+/* global window document */
+
+// eslint-disable-next-line no-undef
```
[the remark this answers is not in the document]
Comment (adamraine) [reply flow-report/standalone-flow.tsx:18]: Eslint claims `LH` is undefined, so typescript eslint may be needed.

**Review by patrickhulce — COMMENTED** [review]
awesome! this is quite palatable to me, but I don't think I'm the one that needed to see the delta to be convinced :)

I'll defer my assigned review to @paulirish for his thoughts

**Comment by paulirish** [comment]
> awesome! this is quite palatable to me, but I don't think I'm the one that needed to see the delta to be convinced :)
> 
> I'll defer my assigned review to @paulirish for his thoughts

seems cool. i'm down.

**Pull request merged**